identity/backend/src/main/java/io/camunda/identity/rolemanagement/service/RoleMembershipService.java

```diff
@@ -0,0 +1,127 @@
+import io.camunda.identity.security.CamundaUserDetails;
+import io.camunda.identity.security.CamundaUserDetailsManager;
+import io.camunda.identity.usermanagement.CamundaGroup;
+import io.camunda.identity.usermanagement.service.GroupService;
+import io.camunda.identity.usermanagement.service.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class RoleMembershipService {
+
+  private final UserService userService;
+  private final GroupService groupService;
+  private final CamundaUserDetailsManager camundaUserDetailsManager;
+  private final RoleRepository roleRepository;
+
+  public RoleMembershipService(
+      final UserService userService,
+      final GroupService groupService,
+      final CamundaUserDetailsManager camundaUserDetailsManager,
+      final RoleRepository roleRepository) {
+    this.userService = userService;
+    this.groupService = groupService;
+    this.camundaUserDetailsManager = camundaUserDetailsManager;
+    this.roleRepository = roleRepository;
+  }
+
+  public List<String> getRolesOfUserByUserId(final long userId) {
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    return camundaUserDetails.getRoles();
+  }
+
+  public void assignRoleToUser(final String roleName, final long userId) {
+    if (!roleRepository.existsById(roleName)) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    final List<String> roles = new ArrayList<>();
+    roles.addAll(camundaUserDetails.getRoles());
+    roles.add(roleName);
+
+    saveUserDetailsWithRoles(camundaUserDetails, roles);
+  }
+
+  public void unassignRoleFromUser(final String roleName, final long userId) {
+    if (!roleRepository.existsById(roleName)) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    final List<String> roles = new ArrayList<>();
+    roles.addAll(
+        camundaUserDetails.getRoles().stream().filter(rn -> !rn.equals(roleName)).toList());
+
+    saveUserDetailsWithRoles(camundaUserDetails, roles);
+  }
+
+  public List<String> getRolesOfGroupByGroupId(final Long groupId) {
```

Comment:
💭 Same as above 
```suggestion
  public List<String> getRolesByGroupId(final Long groupId) {
```

Reply:
You're right, that's still clear and better as it's shorter! Thx! :)